app/src/lib/stores/app-store.ts

```diff
@@ -1024,17 +1025,59 @@ export class AppStore {
   }
 
   /** Update the menu with the names of the user's preferred apps. */
-  private updatePreferredAppMenuItemLabels() {
+  private async updatePreferredAppMenuItemLabels(repository?: Repository) {
     const editorLabel = this.selectedExternalEditor
       ? `Open in ${this.selectedExternalEditor}`
       : undefined
 
+    const prLabel = await this.getPullRequestLabel(repository)
+
     updatePreferredAppMenuItemLabels({
       editor: editorLabel,
+      pullRequestLabel: prLabel,
       shell: `Open in ${this.selectedShell}`,
     })
   }
 
+  private async getPullRequestLabel(repository?: Repository) {
+    if (!repository) {
+      return
+    }
+
+    const githubRepository = repository.gitHubRepository
+
+    if (!githubRepository) {
+      return
+    }
+
+    const repositoryState = this.repositoryState.get(repository.hash)
+
+    if (!repositoryState) {
+      return
+    }
+
+    const pullRequests = await this.pullRequestStore.getPullRequests(
```

Comment:
:lipstick: we can move this down and only do the work if there's a valid branch

Reply:
We shouldn't need to do this at all, really. We have `currentPullRequest` in the app state already.